From c7b27f8e77e673aa2cb2b9be69f908c464ab5d91 Mon Sep 17 00:00:00 2001
From: Pavel Rojtberg <rojtberg@gmail.com>
Date: Mon, 10 May 2021 20:23:46 +0200
Subject: [PATCH] ovis: allow using OGRE devel

only compatible versions support CMake CONFIG mode.
Use that to make check more robust.
---
 modules/ovis/CMakeLists.txt | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/modules/ovis/CMakeLists.txt b/modules/ovis/CMakeLists.txt
index dc190da980b..912b86a590d 100644
--- a/modules/ovis/CMakeLists.txt
+++ b/modules/ovis/CMakeLists.txt
@@ -1,11 +1,11 @@
 set(the_description "OGRE 3D Visualiser.")
 
-find_package(OGRE QUIET)
+find_package(OGRE QUIET CONFIG)
 
 if(NOT OGRE_FOUND)
   message(STATUS "Module opencv_ovis disabled because OGRE3D was not found")
   ocv_module_disable(ovis)
-elseif(OGRE_VERSION VERSION_LESS 1.11.5 OR OGRE_VERSION VERSION_GREATER 2.0)
+elseif(OGRE_VERSION VERSION_LESS 1.11.5)
   message(STATUS "Module opencv_ovis disabled because of incompatible OGRE3D version (${OGRE_VERSION})")
   ocv_module_disable(ovis)
 else() # we need C++11 for OGRE 1.11
